### PR TITLE
Show dynamic greeting message

### PR DESCRIPTION
**Pull request by anh2111htd**
image
image

### CODE REVIEW

File: portfolio/pom.xml [hunk 1]
```diff
@@ -23,6 +23,11 @@
       <version>4.0.1</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+			<groupId>com.google.code.gson</groupId>
+			<artifactId>gson</artifactId>
+			<version>2.8.6</version>
+    </dependency>
```
Comment (rashikaranpuria) [inline portfolio/pom.xml:30]: Resolve indentation

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -23,10 +24,15 @@
...
+		"Hello, good afternoon!",
+		"Have a lovely evening!"
+	};
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.setContentType("application/json;");
+    response.getWriter().println(new Gson().toJson(messages));
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:36]: We do not want to create a new object of Gson everytime this method is called. Consider moving Gson object instantiation out of this method.

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -8,6 +8,7 @@
   </head>
   <body>
     <div id="content">
+      <h1 id="greeting-container"></div>
```
Comment (rashikaranpuria) [inline portfolio/src/main/webapp/index.html:11]: Mismatched starting and ending html tags

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -26,3 +26,16 @@ function addRandomGreeting() {
...
+ */
+function getGreeting() {
+  fetch('/data').then(response => response.json()).then((messages) => {
+    const hour = new Date().getHours();
+    const message = messages[(hour < 12) ? 0 : (hour < 17) ? 1 : 2];
+    document.getElementById('greeting-container').innerText = message;
+  });
+}
+
+getGreeting();
```
Comment (rashikaranpuria) [inline portfolio/src/main/webapp/script.js:41]: nit, blank line

**Comment by anh2111htd** [comment]
image

@rashikaranpuria Hi Rashi, I was not sure how to separate step 3 and step 4, so I decided to work on them in one branch. I will add changes as you suggested for step 3 in this PR as well.

**Pull request merged**